float32 distance_x
float32 distance_y